### PR TITLE
BUG: inconsistency between frame.any/all with dt64 vs dt64tz

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -1640,6 +1640,17 @@ def floor(self, freq, ambiguous="raise", nonexistent="raise"):
     def ceil(self, freq, ambiguous="raise", nonexistent="raise"):
         return self._round(freq, RoundTo.PLUS_INFTY, ambiguous, nonexistent)
 
+    # --------------------------------------------------------------
+    # Reductions
+
+    def any(self, *, axis: Optional[int] = None, skipna: bool = True):
+        # GH#33478
```
Comment (mroeschke) [inline pandas/core/arrays/datetimelike.py:1647]: Is there a reason why this isn't `GH 38723`?

**Review by mroeschke — COMMENTED** [review]
LGTM just one question

File: pandas/core/arrays/datetimelike.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/arrays/datetimelike.py:1647]: typo, should be #33479, which is about the long-term behavior we want, #38723 is about ensuring that the interim behavior is internally-consistent.  just updated

**Review by mroeschke — APPROVED** [review]

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
What is the inconsistency this is fixing? (how is it behaving differently?)

**Comment by jbrockmendel** [comment]
> What is the inconsistency this is fixing? (how is it behaving differently?)

```
dti = pd.date_range("2016-01-01", periods=3)
dti2 = dti.tz_localize("UTC")

df = pd.DataFrame(dti)
df2 = pd.DataFrame(dti2)

>>> df.all()
0    True
dtype: bool

>>> df2.all()
Series([], dtype: bool)
```

**Comment by jorisvandenbossche** [comment]
Thanks for that example, that helps ;) 
(in general, it would helpif you try give a bit more context when opening PRs)

That's indeed certainly an inconsistency to fix. But quickly checking, this actually seems to have worked fine before, so might be a regression? (and thus should potentially target this for 1.2.1)

File: doc/source/whatsnew/v1.3.0.rst [hunk 2]
```diff
@@ -190,6 +190,7 @@ Datetimelike
 - Bug in :class:`DataFrame` and :class:`Series` constructors sometimes dropping nanoseconds from :class:`Timestamp` (resp. :class:`Timedelta`) ``data``, with ``dtype=datetime64[ns]`` (resp. ``timedelta64[ns]``) (:issue:`38032`)
 - Bug in :meth:`DataFrame.first` and :meth:`Series.first` returning two months for offset one month when first day is last calendar day (:issue:`29623`)
 - Bug in constructing a :class:`DataFrame` or :class:`Series` with mismatched ``datetime64`` data and ``timedelta64`` dtype, or vice-versa, failing to raise ``TypeError`` (:issue:`38575`)
+- Bug in :meth:`DataFrame.any` and :meth:`DataFrame.all` behaving differently for tznaive vs tzaware ``datetime64`` dtypes (:issue:`38723`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:193]: move to 1.2.1

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by simonjayhawkins** [comment]
> this actually seems to have worked fine before, so might be a regression?

first bad commit: [50ae0bfc383b5cefa39e82c0446dce11d4ce10e9] BUG: DataFrame reductions inconsistent with Series counterparts (#37827)

File: doc/source/whatsnew/v1.3.0.rst (hunk 2 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v1.3.0.rst]: updated+green

File: doc/source/whatsnew/v1.2.1.rst [hunk 3]
```diff
@@ -17,6 +17,7 @@ Fixed regressions
 - The deprecated attributes ``_AXIS_NAMES`` and ``_AXIS_NUMBERS`` of :class:`DataFrame` and :class:`Series` will no longer show up in ``dir`` or ``inspect.getmembers`` calls (:issue:`38740`)
 - :meth:`to_csv` created corrupted zip files when there were more rows than ``chunksize`` (issue:`38714`)
 - Bug in repr of float-like strings of an ``object`` dtype having trailing 0's truncated after the decimal (:issue:`38708`)
+- Bug in :meth:`DataFrame.any` and :meth:`DataFrame.all` behaving differently for tznaive vs tzaware ``datetime64`` dtypes (:issue:`38723`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.2.1.rst:20]:
```suggestion
- Fixed regression in :meth:`DataFrame.any` and :meth:`DataFrame.all` not returning a result for tz-aware ``datetime64`` columns (:issue:`38723`)
```

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]
I suppose this will also enable the any/all methods on Series? 
(in which case we should add tests for that)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@meeseeksdev backport 1.2.x

**Pull request merged**